tests/shell_tests.bzl

```diff
@@ -83,26 +83,29 @@ def _shell_spawn_e2e_test_impl(ctx):
         "line tab\tcharacter $foo qu\"o\"te it'\\''s foo\\bar back`echo q`uote'",
         '[[ "${output}" == "${expected}" ]]',
     ])
-    script_file = ctx.actions.declare_file("%s.sh" % (ctx.label.name))
     ctx.actions.write(
-        output = script_file,
+        output = ctx.outputs.out,
```

Comment:
Can you check if `ctx.outputs.out` was set? If not, call `declare_file`.

Reply:
I made the attribute mandatory instead.

Reply:
What's the point of the macro `shell_args_test_gen`?

Reply:
To provide a default value for `out`.
`attr.output.default` is deprecated: https://docs.bazel.build/versions/master/skylark/lib/attr.html#output

Reply:
So if you do what I suggested (declare_file is the attribute is under), you can get rid of the macro?

If you think the default value will alwaysnbe good enough, you can actually get rid of the attribute.

Reply:
Done.